index.html

```diff
@@ -174,95 +180,122 @@
       </div>
       <section>
         <h2>
-          Syntax
+          Validation
         </h2>
         <p>
-          When URLs are used for payment method identifiers they MUST be
-          absolute URLs including, at most, a <a>scheme</a>, <a>host</a>,
-          <a>port</a> and <a>path</a>. The URL must be a <a>potentially
-          trustworthy URL</a> as defined in the [[SECURE-CONTEXTS]]
-          specification. Identifier URLs MUST have null <a>query</a> and
-          <a>fragment</a> values.
+          The steps to <dfn data-lt="validate|valid">validate a URL-based
+          payment method identifier</dfn> are given by the following algorithm.
+          The algorithm takes a <var>input</var> string as input and returns
+          either true or an error:
         </p>
+        <ol class="algorithm">
+          <li>Let <var>url</var> be the result of running the <a>URL parser</a>
+          on <var>input</var>.
+          </li>
+          <li>If failure, <a>throw</a> a <code>TypeError</code> exception.
+          </li>
+          <li>If running <a>potentially trustworthy URL</a> with <var>url</var>
+          returns <code>"Not Trustworthy"</code>, <a>throw</a> a
+          <code>TypeError</code> exception.
+          </li>
+          <li>If <var>url</var>'s <a>query</a> is not null or <var>url</var>'s
+          <a>fragment</a> is not null, <a>throw</a> <code>TypeError</code>
+          exception.
+          </li>
+          <li>Return true.
+          </li>
+        </ol>
       </section>
       <section>
         <h2>
           Matching
         </h2>
         <p>
-          User agents MUST use the following algorithm to determine whether two
-          payment method identifiers that are URLs match:
+          The steps to <dfn>determine whether two payment method identifiers
+          match</dfn> is given by the following algorithm. The algorithm takes
+          two string <var>inputA</var> and <var>inputB</var>.
         </p>
-        <ul>
-          <li>Let <em>A</em> be the first payment method identifier string and
-          let <em>B</em> be the second payment method identifier string.
+        <ol class="algorithm">
+          <li>Let <var>urlA</var> be the result of running the <a>URL
+          parser</a> on <var>inputA</var>.
```

Comment:
I actually think we should drop this section entirely. To compare two PMI URLs, you need to have first gone through the validation steps to get `urlA` and `urlB`. If we make that an invariant, then URL spec's "equals" algorithm can just be used without needing any of this.